token/src/instruction.rs

```diff
@@ -26,23 +31,33 @@ pub enum TokenInstruction {
     ///
     /// # Accounts expected by this instruction:
     ///
-    ///   0. `[writable, signer]` New mint to create.
+    ///   0. `[writable, signer]` Mint account to initialize
     ///   1.
     ///      * If supply is non-zero: `[writable]` Account to hold all the newly minted tokens.
     ///      * If supply is zero: `[]` Owner of the mint.
-    ///   2. Optional: `[]` Owner of the mint if supply is non-zero, if present then the 
+    ///   2. Optional: `[]` Owner of the mint if supply is non-zero, if present then the
     ///      token allows further minting of tokens.
     InitializeMint(TokenInfo),
     /// Initializes a new account.  The new account can either hold tokens or be a delegate
     /// for another account.
     ///
     /// # Accounts expected by this instruction:
     ///
-    ///   0. `[writable, signer]`  New account being created.
+    ///   0. `[writable, signer]`  Account to initialize
     ///   1. `[]` Owner of the new account.
     ///   2. `[]` Token this account will be associated with.
     ///   3. Optional: `[]` Source account that this account will be a delegate for.
     InitializeAccount,
+    /// Initializes a multisignature account with N provided signers.
+    /// Multisignature accounts can take the place of an "Owner" account in any token instructions that
+    /// require an owner to be a signer.  The variant field represents the number of required
+    /// signers (M).
+    ///
+    /// # Accounts expected by this instruction:
+    ///
+    ///   0. `[signer]` Multisignature account to initialize
+    ///   1-11. `[]` Signer accounts, must equal to N where 1 <= N <= 11
+    InitializeMultisig(u8),
```

Comment:
wdyt about including N in the instruction data: 
```suggestion
    InitializeMultisig {n: u8, m: u8},
```
N is implicit by the number of signer accounts provided but for the cost of an extra 8 bits, adding an explicit N seems to be more readable

Reply:
If we do it's adding a possible failure point and more work internally to check that the user didn't mess it up (though on the other hand, it does add redundancy to make sure the user included the intended number of singers to the instruction